### PR TITLE
feat: add link

### PR DESCRIPTION
**Pull request by tanvibhakta**
## Description

Zeplin: https://zpl.io/o1y8OEZ
Finishes https://github.com/appsmithorg/appsmith/issues/19345

## Type of change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## How Has This Been Tested?

- Manual on storybook 
- Manual on main repo
- Jest
- Cypress

### Issues raised during DP testing
https://github.com/appsmithorg/design-system/pull/351#issuecomment-1425761580
https://github.com/appsmithorg/design-system/pull/351#issuecomment-1428135978

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: packages/design-system/src/Link/Link.styles.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    }
+  `,
+};
+export const StyledLink = styled(RouterLink)<{
+  kind?: Kind;
+}>`
+  ${Variables}
+
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
```
Comment (albinAppsmith) [inline packages/design-system/src/Link/Link.styles.tsx:30]: Why we are suppressing error here?

File: packages/design-system/src/Link/Link.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  return (
+    <StyledLink
+      {...linkProps}
+      innerRef={ref}
+      kind={props.kind}
+      target={props.target}
+      to={props.to}
+    >
+      {props.endIcon && <StyledIcon name={props.endIcon} size="1rem" />}
```
Comment (albinAppsmith) [inline packages/design-system/src/Link/Link.tsx:34]: start icon is placed in the end and end icon is placed in start

**Review by albinAppsmith — APPROVED** [review]

**Comment by shadabbuchh** [comment]
Bugs in this PR:
**Functional:**

1. If we add the value `_self` in the `target` control nothing happens.

**Design:**

1. Incorrect padding for start & end icons.

**Question:**
Do you think we should display an underline when we hover/click over the component?

**Comment by tanvibhakta** [comment]
> If we add the value _self in the target control nothing happens.

It looks like React Router no longer handles _self appropriately: https://github.com/remix-run/react-router/issues/6750
(some interesting reading for why target is the way it is https://stackoverflow.com/questions/18470097/difference-between-self-top-and-parent-in-the-anchor-tag-target-attribute)

> external urls

I've now added support for an `a` tag to be rendered. This is in a different story; you will not be able to test this feature in the first story due to some limitations with storybook.

>  Incorrect padding for start & end icons.

Added.

>  Do you think we should display an underline when we hover/click over the component?

I personally would go for something like this, but I believe design has made the decision to remove this. I suppose the underline pattern is more common in an app with lots of text and links together, as opposed to an SPA like ours?

**Comment by shadabbuchh** [comment]
New issues:

1. For External link when we add an external link and do not include `https://` & instead straightaway add `www.` to it, we get the below error. 
Screenshot 2023-02-13 at 1 50 08 PM

If the component is accepting a value from user, we can’t enforce user to enter `https://` and then it will cause an issue.

2. Also now an underline has been introduced in the component upon hover/click. Please refer the design.

**Comment by tanvibhakta** [comment]
urls without a header are not supported by the `href` property - see:https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-href

The above link contains a playground with which you can test a "www.example.com" site and see how it breaks as well. Links must contain a domain name (https, mailto, etc) for them to be considered absolute links, otherwise the browser will attempt to append the href value to the current domain. 

See https://stackoverflow.com/a/8998176/4087799 to read more.

Design has made the decision to not go with the underlines for now.

**Pull request merged**